pkg/oc/cli/admin/prune/deployments/prune.go

```diff
@@ -111,19 +110,6 @@ func NewDeploymentDeleter(deployments corev1client.ReplicationControllersGetter,
 
 func (p *deploymentDeleter) DeleteDeployment(deployment *corev1.ReplicationController) error {
 	glog.V(4).Infof("Deleting deployment %q", deployment.Name)
-	// If the deployment is failed we need to remove its deployer pods, too.
-	if appsutil.IsFailedDeployment(deployment) {
-		dpSelector := appsutil.DeployerPodSelector(deployment.Name)
-		deployers, err := p.pods.Pods(deployment.Namespace).List(metav1.ListOptions{LabelSelector: dpSelector.String()})
-		if err != nil {
-			glog.Warningf("Cannot list deployer pods for %q: %v\n", deployment.Name, err)
-		} else {
-			for _, pod := range deployers.Items {
-				if err := p.pods.Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
-					glog.Warningf("Cannot remove deployer pod %q: %v\n", pod.Name, err)
-				}
-			}
-		}
-	}
-	return p.deployments.ReplicationControllers(deployment.Namespace).Delete(deployment.Name, nil)
+	policy := metav1.DeletePropagationBackground
```

Comment:
technically it was foreground before, but background seems more reasonable, especially given that Pods are actually deleted by kubelet which mean it could hand if a Node was down